File: src/fiefapp.svelte/src/models/fief.ts
import type { Port } from './port';
import type { Building } from './building';
import type { Boat } from './boat';
import type { Resident } from './resident';
import type { Soldier } from './soldier';
import type { Employee } from './employee';
import type { Builder } from './builder';
import type { Industry } from './industries/industry';
import { Village } from './village';
import { Inheritance } from './inheritance';
import { Road } from './road';
import { Livingcondition } from './livingcondition';
import { Market } from './market';

interface IFief {
  id: string;
  gamesessionId: string;
  playerId: string | undefined;
  name: string;
  manorName: string;
  acres: number;
  farmlandAcres: number;
  pastureAcres: number;
  woodlandAcres: number;
  fellingAcres: number;
  unusableAcres: number;
  animalHusbandryQuality: number;
  agriculturalQuality: number;
  fishingQuality: number;
  oreQuality: number;
  huntingQuality: number;
  animalHusbandryDevelopmentLevel: number;
  agriculturalDevelopmentLevel: number;
  fishingDevelopmentLevel: number;
  woodlandDevelopmentLevel: number;
  oreDevelopmentLevel: number;
  huntingDevelopmentLevel: number;
  educationDevelopmentLevel: number;
  healthcareDevelopmentLevel: number;
  militaryDevelopmentLevel: number;
  seafaringDevelopmentLevel: number;
  market?: Market;
  stewardId?: string;
  port?: Port;
  livingcondition: Livingcondition;
  road: Road;
  inheritance: Inheritance;
  villages: Village[];
  industries: Industry[];
  buildings: Building[];
  boats: Boat[];
  residents: Resident[];
  soldiers: Soldier[];
  employees: Employee[];
  builders: Builder[];
}

export interface IFiefs {
  fiefs: IFief[];
}

export interface IShortFief {
  id: string;
  name: string;
}

export class Fief implements IFief {
  id: string;
  gamesessionId: string;
  playerId: string | undefined;
  name: string;
  manorName: string;
  acres: number;
  farmlandAcres: number;
  pastureAcres: number;
  woodlandAcres: number;
  fellingAcres: number;
  unusableAcres: number;
  animalHusbandryQuality: number;
  agriculturalQuality: number;
  fishingQuality: number;
  oreQuality: number;
  huntingQuality: number;
  animalHusbandryDevelopmentLevel: number;
  agriculturalDevelopmentLevel: number;
  fishingDevelopmentLevel: number;
  woodlandDevelopmentLevel: number;
  oreDevelopmentLevel: number;
  huntingDevelopmentLevel: number;
  educationDevelopmentLevel: number;
  healthcareDevelopmentLevel: number;
  militaryDevelopmentLevel: number;
  seafaringDevelopmentLevel: number;
  market?: Market;
  stewardId?: string;
  port?: Port;
  livingcondition: Livingcondition;
  road: Road;
  inheritance: Inheritance;
  villages: Village[];
  industries: Industry[];
  buildings: Building[];
  boats: Boat[];
  residents: Resident[];
  soldiers: Soldier[];
  employees: Employee[];
  builders: Builder[];

  constructor(name?: string, playerId?: string, gamesessionId?: string) {
    this.id = '';
    this.gamesessionId = gamesessionId ? gamesessionId : '';
    this.name = name ? name : 'Ny förläning';
    this.playerId = playerId ? playerId : undefined;
    this.manorName = 'Ny förlänings gods';
    this.acres = 0;
    this.farmlandAcres = 0;
    this.pastureAcres = 0;
    this.woodlandAcres = 0;
    this.fellingAcres = 0;
    this.unusableAcres = 0;
    this.animalHusbandryQuality = 1;
    this.agriculturalQuality = 1;
    this.fishingQuality = 1;
    this.oreQuality = 1;
    this.huntingQuality = 1;
    this.animalHusbandryDevelopmentLevel = 1;
    this.agriculturalDevelopmentLevel = 1;
    this.fishingDevelopmentLevel = 1;
    this.woodlandDevelopmentLevel = 1;
    this.oreDevelopmentLevel = 1;
    this.huntingDevelopmentLevel = 1;
    this.educationDevelopmentLevel = 1;
    this.healthcareDevelopmentLevel = 1;
    this.militaryDevelopmentLevel = 1;
    this.seafaringDevelopmentLevel = 1;
    this.market = new Market();
    this.stewardId = undefined;
    this.port = undefined;
    this.livingcondition = new Livingcondition();
    this.road = new Road();
    this.inheritance = new Inheritance();
    this.villages = [new Village()];
    this.industries = [];
    this.buildings = [];
    this.boats = [];
    this.residents = [];
    this.soldiers = [];
    this.employees = [];
    this.builders = [];
  }
}

File: src/fiefapp.entities/Fief.cs
using fiefapp.entities.BaseClasses;
using fiefapp.entities.Interfaces;
using System;
using System.Collections.Generic;

namespace fiefapp.entities
{
    public class Fief : IBaseEntity, INeedSteward
    {
        public Fief()
        {
            Id = Guid.NewGuid();
            GamesessionId = Guid.NewGuid();
            Name = "Ny förläning";
            ManorName = "Ny förlänings högsäte";
        }

        public Guid Id { get; set; }
        public Guid GamesessionId { get; set; }
        public string UserId { get; set; }
        public string Name { get; set; }
        public string ManorName { get; set; }
        public int Acres { get; set; }
        public int FarmlandAcres { get; set; }
        public int PastureAcres { get; set; }
        public int WoodlandAcres { get; set; }
        public int FellingAcres { get; set; }
        public int UnusableAcres { get; set; }
        public Guid? MarketId { get; set; }
        public Market Market { get; set; }
        public Guid? StewardId { get; set; }
        public Steward Steward { get; set; }
        public Guid? PortId { get; set; }
        public Port Port { get; set; }
        public Guid LivingconditionId { get; set; }
        public Livingcondition Livingcondition { get; set; }
        public Guid RoadId { get; set; }
        public Road Road { get; set; }
        public Guid InheritanceId { get; set; }
        public Inheritance Inheritance { get; set; }
        public ICollection<Guid> VillageIds { get; set; }
        public ICollection<Village> Villages { get; set; }
        public ICollection<Guid> IndustryIds { get; set; }
        public ICollection<Industry> Industries { get; set; }
        public ICollection<Guid> BuildingIds { get; set; }
        public ICollection<Building> Buildings { get; set; }
        public ICollection<Guid> ResidentIds { get; set; }
        public ICollection<Resident> Residents { get; set; }
        public ICollection<Guid> SoldierIds { get; set; }
        public ICollection<Soldier> Soldiers { get; set; }
        public ICollection<Guid> EmployeeIds { get; set; }
        public ICollection<Employee> Employees { get; set; }
        public ICollection<Guid> BuilderIds { get; set; }
        public ICollection<Builder> Builders { get; set; }
    }
}